src/main/java/org/springframework/integration/aws/inbound/kinesis/KinesisMessageDrivenChannelAdapter.java

```diff
@@ -340,6 +342,16 @@ public void setBindSourceRecord(boolean bindSourceRecord) {
 		this.bindSourceRecord = bindSourceRecord;
 	}
 
+	/**
+	 * Set to false to disable automatically writing a checkpoint of the {@code endingSequenceNumber}
+	 * for a shard which is closed and has had all of its records read
+	 *
+	 * @param checkpointClosedShards false to disable automatic checkpointing at shard end
+	 */
+	public void setCheckpointClosedShards(boolean checkpointClosedShards) {
```

Comment:
I don't think we need this option at all.
As far as you say we are in a loop reading closed shard for nothing, so we definitely should not deal with them at all.
Since you have provided checkpointing in the fix, we are good to go with that agreed behavior from now on.